podkrepi-bg/frontend
# Allow the user to create recurring donations per campaign

## Motivation and context

Some users want to be able to make monthly donations automatically.
Now they can subscribe to a campaign, instead of a one-time donation.
I've added a checkbox on the donations page. 
If this checkbox is selected, a subscription is created in stripe, instead of a payment.
The user can also cancel a recurring donation from its profile.

## Screenshots:

![image](https://user-images.githubusercontent.com/99622/206322865-e3e661f6-ab0c-4eed-abe6-db6e34c251ac.png)

![image](https://user-images.githubusercontent.com/99622/206322910-4f0bf0fe-463a-4077-9b5c-5085e47a39d1.png)


![image](https://user-images.githubusercontent.com/99622/206322764-132d17a8-9d45-47ac-b05c-7a910de32601.png)




Affected pages: 
All the steps in the donations dialog.
The users profile.
Admins recurring donations page.

## Testing

Click on the checkbox and you should be redirected to a subscription instead of a payment.

### Steps to test
1. Open a campaign.
2. Click donate
3. Choose a stripe sum
4. Click on the "Donate the same amount every month..." checkbox
5. Follow the process all the way until you are redirected to stripe.
6. There you should see a subscription option instead of the payment one.

### Affected urls

We need to add at least one tiered subscription price with a value of 1 BGN in the stripe account.
It would be best if we add one subscription price per each one-time price. This would make it slightly more user-friendly.







## Environment

### New environment variables:



### New or updated dependencies:
